examples/gno.land/p/demo/auction/auction.gno

```diff
@@ -0,0 +1,140 @@
+package auction
+
+import (
+	"std"
+	"strconv"
+	"time"
+
+	"gno.land/p/demo/avl"
+)
+
+// Main struct
+type Auction struct {
+	title       string
+	owner       std.Address
+	description string
+	begin       time.Time
+	end         time.Time
+	price       uint64
+	bids        *avl.Tree // key: std.Address, value: Bid
+}
+
+type Bid struct {
+	bidder std.Address
+	amount uint64
+}
+
+type EventEnd struct {
+	winner std.Address
+	amount uint64
+}
+
+func NewAuction(
+	title string,
+	owner std.Address,
+	description string,
+	begin time.Time,
+	end time.Time,
+	minPrice uint64,
+) *Auction {
+	auction := &Auction{
+		title:       title,
+		description: description,
+		owner:       owner,
+		bids:        avl.NewTree(),
+		begin:       begin,
+		end:         end,
+		price:       minPrice,
+	}
+	// Emit AuctionStart event
+	std.Emit("AuctionStart", "title", title, "time", begin.String())
+	return auction
+}
+
+// GetTitle returns the title of the auction.
+func (a Auction) GetTitle() string {
+	return a.title
+}
+
+// GetDescription returns the description of the auction.
+func (a Auction) GetDescription() string {
+	return a.description
+}
+
+// GetOwner returns the address of the auction owner.
+func (a Auction) GetOwner() std.Address {
+	return a.owner
+}
+
+// GetBegin returns the start time of the auction.
+func (a Auction) GetBegin() time.Time {
+	return a.begin
+}
+
+// GetEnd returns the end time of the auction.
+func (a Auction) GetEnd() time.Time {
+	return a.end
+}
+
+// GetPrice returns the current highest bid amount.
+func (a Auction) GetPrice() uint64 {
+	return a.price
+}
+
+// GetBids returns an array of all the bids made in the auction.
+func (a Auction) GetBids() []*Bid {
+	bids := make([]*Bid, 0, a.bids.Size())
+	a.bids.Iterate("", "", func(key string, value interface{}) bool {
```

Comment:
What do you think about adding some kind of pagination here, to circumvent an execution attack if there are many bids?